OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -71,7 +71,12 @@ public override void OnBecomingIdle(Actor self)
 			if (withSpriteBody.DefaultAnimation.CurrentSequence.Name != Info.IdleSequence)
 				withSpriteBody.DefaultAnimation.PlayRepeating(Info.IdleSequence);
 
-			base.OnBecomingIdle(self);
+			var thumper = self.World.FindActorsInCircle(self.CenterPosition, Info.MaxSearchRadius)
```

Comment:
This doesn't account for proximity or multiple thumpers, so the worm will behave strangely if you have more than one in an area.